aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -819,25 +820,80 @@ Tensor& randn_out(IntArrayRef size, c10::optional<Generator> generator, Tensor&
   return result.normal_(0, 1, generator);
 }
 
-Tensor normal(double mean, double std, IntArrayRef size,
-              c10::optional<Generator> generator,
-    c10::optional<ScalarType> dtype,
-    c10::optional<Layout> layout,
-    c10::optional<Device> device,
-    c10::optional<bool> pin_memory) {
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ normal ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Tensor normal(
+  double mean,
+  double std,
+  IntArrayRef size,
+  c10::optional<Generator> generator,
+  c10::optional<ScalarType> dtype,
+  c10::optional<Layout> layout,
+  c10::optional<Device> device,
+  c10::optional<bool> pin_memory
+) {
+  CHECK_NORMAL_STD(std);
+
   // See [Note: hacky wrapper removal for TensorOptions]
-  TensorOptions options = TensorOptions().dtype(dtype).layout(layout).device(device).pinned_memory(pin_memory);
+  TensorOptions options = TensorOptions()
+    .dtype(dtype)
+    .layout(layout)
+    .device(device)
+    .pinned_memory(pin_memory);
 
   auto result = at::empty(size, options);
   return result.normal_(mean, std, generator);
 }
 
-Tensor& normal_out(double mean, double std,
-                   IntArrayRef size, c10::optional<Generator> generator, Tensor& result) {
+Tensor normal_meta(
+  double mean,
+  double std,
+  IntArrayRef size,
+  c10::optional<Generator> generator,
+  c10::optional<ScalarType> dtype,
+  c10::optional<Layout> layout,
+  c10::optional<Device> device,
+  c10::optional<bool> pin_memory
+) {
+  CHECK_NORMAL_STD(std);
+
+  // See [Note: hacky wrapper removal for TensorOptions]
+  TensorOptions options = TensorOptions()
+    .dtype(dtype)
+    .layout(layout)
+    .device(device)
+    .pinned_memory(pin_memory);
+
+  auto result = at::empty(size, options);
+  return result;  // similar to normal_meta_
+}
+
+Tensor& normal_out(
```

Comment:
It seems to me that both `normal_out` and `normal_out_meta` do the same thing. You could dispatch CPU, CUDA, and META to the same function.